modular-framework/list.h

```diff
@@ -0,0 +1,620 @@
+#ifndef _LINUX_LIST_H
+#define _LINUX_LIST_H
+
+#include <stddef.h>
+
+#if defined(__STDC__)
+#define typeof __typeof__
+#endif
+
```

Comment:
Yes, I'm changing directory structure as:
odp/frameworks/modular/

So laterly odp/frameworks/ toplevel directory can hold other frameworks like device driver framework focusing on providing facilities to support driver development.